src/js/pages/jobs/JobDetailPage.js

```diff
@@ -57,26 +72,32 @@ class JobDetailPage extends mixin(StoreMixin, TabsMixin) {
   }
 
   handleEditButtonClick() {
-    // TODO: Handle edit button click
+    this.setState({isJobFormModalOpen: true});
+  }
+
+  handleCloseJobFormModal() {
+    this.setState({isJobFormModalOpen: false});
   }
 
-  handleRunNowButtonClick(jobID) {
-    ChronosStore.runJob(jobID);
+  handleRunNowButtonClick() {
+    let job = ChronosStore.getJob(this.props.params.id);
+
+    ChronosStore.runJob(job.getId());
   }
 
-  handleMoreDropdownSelection(jobID, selection) {
+  handleMoreDropdownSelection(selection) {
+    let job = ChronosStore.getJob(this.props.params.id);
+
     if (selection.id === 'suspend') {
-      ChronosStore.suspendJob(jobID);
+      ChronosStore.suspendJob(job.getId());
     }
 
     if (selection.id === 'destroy') {
-      ChronosStore.deleteJob(jobID);
+      ChronosStore.deleteJob(job.getId());
```

Comment:
I'd have preferred to see these various fixes in a separate PR but I won't complain 😂